File: cobots_core/src/cobots_model/data/location.py
from waypoint import Waypoint


class Location(Waypoint):

    '''
    Data structure methods
    '''

    def __init__(self, position=None, orientation=None, joints=None, type='',
                 name='', uuid=None, parent=None, append_type=True):
        super(Location,self).__init__(
            position=position,
            orientation=orientation,
            joints=joints,
            type='location.'+type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type)

File: evd_authoring_app/src/model/evd_script/data/location.js
import { Waypoint } from './waypoint';
import { Position, Orientation } from './geometry';

export class Location extends Waypoint {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'location.';
    }

    static fullTypeString() {
        return Waypoint.fullTypeString() + Location.typeString();
    }

    constructor(position=null, orientation=null, joints=null, type='', name='', uuid=null, parent=null, appendType=true) {
        super(
            position= position,
            orientation= orientation,
            joints= joints,
            type= (appendType) ? 'location.'+type : type,
            name= name,
            uuid= uuid,
            parent= parent,
            appendType= appendType
        );
    }

    static fromDict(dct) {
        return new Location(
            position= Position.fromDict(dct.position),
            orientation= Orientation.fromDict(dct.orientation),
            joints= dct.joints,
            type= dct.type,
            name= dct.name,
            uuid= dct.uuid,
            appendType= false
        );
    }

    toBlockly() {
        //TODO implement this
        return {};
    }
}